### PR TITLE
Add new examples

### PR DESCRIPTION
**Pull request by lxrobotics**

### CODE REVIEW

File: examples/MKRWIFI1010_Cloud_Blink/MKRWIFI1010_Cloud_Blink.ino [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ConnectionManager *ArduinoIoTPreferredConnection = new WiFiConnectionManager(SECRET_WIFI_NAME, SECRET_PASSWORD);
+
+void setup() {
+  setDebugMessageLevel(3); // used to set a level of granularity in information output [0...4]
+  Serial.begin(9600);
+  while (!Serial); // waits for the serial to become available
+  ArduinoCloud.begin(ArduinoIoTPreferredConnection); // initialize a connection to the Arduino IoT Cloud
+  while (ArduinoCloud.connected()); // needed to wait for the inizialization of CloudSerial
+  CloudSerial.print("I'm ready for blinking!\n");
+}
```
Comment (lxrobotics) [inline examples/MKRWIFI1010_Cloud_Blink/MKRWIFI1010_Cloud_Blink.ino]:
```suggestion
  pinMode(LED_BUILTIN, OUTPUT);
}
```

**Comment by ubidefeo** [comment]
Because of the different init process for GSM and its required parameters we should have a separate example.
I've tried to tackle a singe example encompassing all supported boards but the excess of `#ifdef` made me refrain from it :)
@umbynos any comments?

File: examples/MKRWIFI1010_Cloud_Blink/MKRWIFI1010_Cloud_Blink.ino [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+String cloudSerialBuffer = ""; // the string used to compose network messages from the received characters
+// handles connection to the network
+ConnectionManager *ArduinoIoTPreferredConnection = new WiFiConnectionManager(SECRET_WIFI_NAME, SECRET_PASSWORD);
+
+void setup() {
+  setDebugMessageLevel(3); // used to set a level of granularity in information output [0...4]
+  Serial.begin(9600);
+  while (!Serial); // waits for the serial to become available
+  ArduinoCloud.begin(ArduinoIoTPreferredConnection); // initialize a connection to the Arduino IoT Cloud
+  while (ArduinoCloud.connected()); // needed to wait for the inizialization of CloudSerial
```
Comment (per1234) [inline examples/MKRWIFI1010_Cloud_Blink/MKRWIFI1010_Cloud_Blink.ino]:
```suggestion
  while (ArduinoCloud.connected()); // needed to wait for the initialization of CloudSerial
```

**Comment by lxrobotics** [comment]
@ubidefeo @umbynos I've patched everything up so the file compiles also with travis for all boards. Two things:

- In order to compile for all boards (and especially to cover undefined secrets) we need quite a bit of `#ifdef` magic - we could extract that into another file (e.g. `arduino_travis_secrets.h`). 
- Since the board works for both GSM and WIFI enabled boards I'd propose to rename the example simply to `Cloud_Blink`. 

I'd love to hear your opinion on this:smiley:

**Comment by umbynos** [comment]
This sketch will be the one used in the getting started procedure. I've started a rework in order to use the new connection flow. So I think having one sketch for 1000, 1010 and GSM is a good thing, but maybe we should hear an opinion from @AlbyIanna.
@ubidefeo Originally there were different sketches for different boards. One for 1010, one for GSM and so on.
@lxrobotics `Cloud_Blink` sounds good to me.

**Comment by umbynos** [comment]
Sorry about the mess I did just above. I'm still a noob using GitHub :sweat_smile: 
I split `MKRWIFI1010_Cloud_Blink` in two examples:
* `WiFi_Cloud_Blink` which is compatible with MKR1010 and MKR1000
* `GSM_Cloud_Blink` which is compatible with MKRGSM1400
I did this beacuse on create all the secrets are parsed and it's not possible to hide them. So having a single sketch will generate all the secrets: some for WiFi boards and some for GSM boards.
Although I'm still having some problems with the GSM sketch, I don't know why the serial monitor remains empty.

**Comment by lxrobotics** [comment]
Good Morning @umbynos,

Don't worry about the mess, it can always be cleaned up by applying `git rebase` :wink: . I'll send you a confluence article which I wrote on this topic via slack.

Regardless of this I currently see two problems with this PR:

- Since `GSM_Cloud_Blink.ino` and `WiFi_Cloud_Blink.ino` contain platform specific code for the MKR GSM 1400 and the MKR WIFI 1010 they can not be successfully run by our travis build script, due to the fact that the build script build each example against all platforms. @per1234 Do you have an idea how to modify the travis build script so that `GSM_Cloud_Blink.ino` will be only build for MKR GSM 1400 and `WiFi_Cloud_Blink.ino` will only be build for MKR 1000 and MKR WIFI 1010?
- You are mentioning that you have problems with the serial monitor for the GSM sketch. This sounds to me like this PR is not complete yet. You can contact me on slack and maybe we can sort those issues out together in order to get this PR to completion. For future PRs: If at the time of opening a PR you are aware that there is still work to be done, please use the feature of creating a **draft pull request** ( https://github.blog/2019-02-14-introducing-draft-pull-requests/ ). Doing so ensures that people taking a look at your PR know it's work in progress and they can respond accordingly :wink:

**Comment by umbynos** [comment]
Thank you! I'm going to read the article you sent me right now.
* Now I understand the problem with travis errors, but I don't know how to solve them.
* I did't know about **draft pull request**, but in the future I will certainly be using them!
Thank you for your advises.

**Comment by per1234** [comment]
>@per1234 Do you have an idea how to modify the travis build script so that GSM_Cloud_Blink.ino will be only build for MKR GSM 1400 and WiFi_Cloud_Blink.ino will only be build for MKR 1000 and MKR WIFI 1010?

I came up with two methods by which this could be accomplished:

### Define a separate `script` phase for each job
https://github.com/per1234/ArduinoIoTCloud/commit/7a2eacfb5666641ed4b185c3f6fba20f121fa9af
build:
https://travis-ci.org/per1234/ArduinoIoTCloud/builds/520194410

Cons:
- Some duplication of the `script` phase configuration between each job

### Use conditional logic
https://github.com/per1234/ArduinoIoTCloud/commit/0976b58af4c8fea2ecf1f5a4ea68661848fc249b
build:
https://travis-ci.org/per1234/ArduinoIoTCloud/builds/520184751

Cons:
- The conditional code makes .travis.yml and the job logs a bit "messy".

**Comment by lxrobotics** [comment]
Thank you very much for your input :+1:

In my opinion the first version ( https://github.com/per1234/ArduinoIoTCloud/commit/0976b58af4c8fea2ecf1f5a4ea68661848fc249b ) should be implemented, since it is much simpler to read and understand (by humans). If the downside is that the CI server has a bit more work I'd say that's a small sacrifice for the increased readability (and therefore reduced probability of bug introduction).

**Comment by per1234** [comment]
>In my opinion the first version ( per1234@0976b58 )

Just to avoid any confusion, I should point out that per1234@0976b58 is the commit for the second version.

I don't believe that the per-job `script` definitions will result in any more work for the CI server. My concern with the duplication is that it makes maintenance a bit more work. For example, if a new example sketch is added to this library, it would then need to added to three separate locations in .travis.yml. I don't think that's really a significant problem though, and it's easily balanced out by the benefits of improved readability.

If you are concerned with efficiency of the CI build, a `before_install` phase could be defined for each job. This would allow only installing the libraries needed for that specific job. For example, there is no need to install the WiFi101 or MKRGSM libraries for the `BOARD="arduino:samd:mkrwifi1010"` job.

File: .travis.yml [hunk 3]
```diff
@@ -30,5 +30,7 @@ install:
 script:
   - buildExampleSketch ArduinoIoTCloud_LED_switch
   - buildExampleSketch ArduinoIoTCloud_Travis_CI
+  - buildExampleSketch WiFi_Cloud_Blink
```
Comment (per1234) [inline .travis.yml]:
```suggestion
  - |
    if [ "$BOARD" == "arduino:samd:mkr1000" ] || [ "$BOARD" == "arduino:samd:mkrwifi1010" ];
    then buildExampleSketch WiFi_Cloud_Blink;
    fi
```

File: .travis.yml [hunk 4]
```diff
@@ -30,5 +30,7 @@ install:
...
+  - buildExampleSketch GSM_Cloud_Blink
```
Comment (per1234) [inline .travis.yml]:
```suggestion
  - |
    if [ "$BOARD" == "arduino:samd:mkrgsm1400" ];
    then buildExampleSketch GSM_Cloud_Blink;
    fi
```

**Pull request merged**